tools/importer-msgraph-metadata/components/parser/types.go

```diff
@@ -96,61 +104,136 @@ func (m *Model) IsValid() bool {
 	return true
 }
 
+func (m *Model) DataApiSdkModel(models Models) (*sdkModels.SDKModel, error) {
+	sdkFields := make(map[string]sdkModels.SDKField)
+	for fieldName, field := range m.Fields {
+		objectDefinition, err := field.DataApiSdkObjectDefinition(models)
+		if err != nil {
+			return nil, err
+		}
+
+		if objectDefinition == nil {
+			return nil, fmt.Errorf("could not determine SDKObjectDefinition for field: %s", fieldName)
+		}
+
+		sdkFields[fieldName] = sdkModels.SDKField{
+			ContainsDiscriminatedValue: false,
+			DateFormat:                 nil,
+			Description:                field.Description,
+			JsonName:                   field.JsonField,
+			ObjectDefinition:           *objectDefinition,
+
+			// TODO work these out
+			Optional:  false,
+			ReadOnly:  false,
+			Required:  false,
+			Sensitive: false,
+		}
+	}
+
+	return &sdkModels.SDKModel{
+		DiscriminatedValue:                    nil,
+		FieldNameContainingDiscriminatedValue: nil,
+		Fields:                                sdkFields,
+		ParentTypeName:                        nil,
+	}, nil
+}
+
 type ModelField struct {
 	Title        string
 	Type         *DataType
 	Description  string
 	Default      interface{}
-	Enum         []string
 	ItemType     *DataType
 	ConstantName *string
 	ModelName    *string
 	JsonField    string
 }
 
-// CSType returns a string containing the C# type name for the ModelField, either describing it as a literal type, a
-// specific model type, or a collection of either.
-func (f ModelField) CSType(models Models) *string {
+func (f ModelField) DataApiSdkObjectDefinition(models Models) (*sdkModels.SDKObjectDefinition, error) {
+	if f.ConstantName != nil {
+		return &sdkModels.SDKObjectDefinition{
+			NestedItem:    nil,
+			ReferenceName: f.ConstantName,
+			Type:          sdkModels.ReferenceSDKObjectDefinitionType,
+		}, nil
+	}
+
 	if f.Type == nil {
-		return nil
+		// Default to strings when the type is unknown
```

Comment:
Out of interest how often is the type unknown here?

Reply:
Actually found quite a few occurrences when trying to answer this (and fixed them). I've changed it to a failure state so these can be better surfaced.